src/base/net/reverseresolution.cpp

```diff
@@ -52,37 +52,43 @@ ReverseResolution::ReverseResolution(QObject *parent)
 
 ReverseResolution::~ReverseResolution()
 {
-    qDebug("Deleting host name resolver...");
+    // abort on-going lookups instead of waiting them
+    for (auto iter = m_lookups.cbegin(); iter != m_lookups.cend(); ++iter)
+        QHostInfo::abortHostLookup(iter.key());
 }
 
-void ReverseResolution::resolve(const QString &ip)
+void ReverseResolution::resolve(const QHostAddress &ip)
 {
-    if (m_cache.contains(ip)) {
-        const QString &hostname = *m_cache.object(ip);
-        qDebug() << "Resolved host name using cache: " << ip << " -> " << hostname;
-        if (isUsefulHostName(hostname, ip))
-            emit ipResolved(ip, hostname);
-    }
-    else {
-        // Actually resolve the ip
-        m_lookups.insert(QHostInfo::lookupHost(ip, this, &ReverseResolution::hostResolved), ip);
+    if (ip.isNull())
+        return;
```

Comment:
I would implement some kind of error handling here.
At least it should be consistent with other code paths, i.e. ends up with ipResolved signal to indicate that resolution process I done.

Reply:
Seems current code has the same inconsistency...

Reply:
I added another commit, I'll squash it when you agree with it.